src/sgame/components/SpectatorClassComponent.cpp

```diff
@@ -0,0 +1,35 @@
+#include "SpectatorClassComponent.h"
+
+SpectatorClassComponent::SpectatorClassComponent(Entity& entity, ClientComponent& r_ClientComponent)
+	: SpectatorClassComponentBase(entity, r_ClientComponent)
+{}
+
+void SpectatorClassComponent::HandlePrepareNetCode() {
+	gclient_t* cl = entity.oldEnt->client;
+
+	if (cl && cl->sess.spectatorState == SPECTATOR_FOLLOW) {
```

Comment:
Instead of nesting the three `if` statements, maybe just return early if the respective condition is not met?